### DISCUSSION TITLE
Way to suppress the OpenMetrics types in metrics generated by prometheus exporter

### DISCUSSION THREAD

**Discussion by yzhuang93** [open-telemetry/opentelemetry-java-instrumentation#11282] [Q&A] [answered]
Hi there, we are using opentelemetry java agent version 1.25. The prometheus exporter in this version is exporting metrics following the OpenMetrics format which includes those "Info" types, for example:
```
# TYPE target info
# HELP target Target metadata
target_info{host_arch="amd64",host_name=..........} 1
# TYPE otel_scope_info info
# HELP otel_scope_info Scope metadata
otel_scope_info{otel_scope_name="io.opentelemetry.micrometer-1.5"} 1
# TYPE otel_scope_info info
# HELP otel_scope_info Scope metadata
otel_scope_info{otel_scope_name="io.opentelemetry.exporters.jaeger-grpc-okhttp"} 1
```
I can definitely understand following OpenMetrics spec is good, but this caused some trouble for us as well as we are using telegraf in our system to parse the metrics, and the telegraf has not yet supported these metrics types (Still adapting the old Promethues format)
I am wondering if it is possible to suppress these metrics with 'Info' metrics type, as I saw in https://opentelemetry.io/docs/specs/otel/metrics/sdk_exporters/prometheus/ says we might have the option `without_target_info` and `without_scope_info`, but I am not sure if we have that in java opentelemetry as well, as I don't find any related doc for that.

**Comment by laurit** [comment, marked as answer]
> I am wondering if it is possible to suppress these metrics with 'Info' metrics type, as I saw in https://opentelemetry.io/docs/specs/otel/metrics/sdk_exporters/prometheus/ says we might have the option without_target_info and without_scope_info, but I am not sure if we have that in java opentelemetry as well, as I don't find any related doc for that.

This question is best asked from https://github.com/open-telemetry/opentelemetry-java. If opentelemetry-java does not provide such options then perhaps collector does, if it is so you could send metrics to collector via otlp and use the prometheus exporter in collector.

**Comment by yzhuang93** [comment]
Just for anyone might find this discussion in the future, this issue has been resolved: https://github.com/open-telemetry/opentelemetry-java/issues/5693 which means if you have your opentelemetry java agent with version higher than 2.0.0, the target info type has been converted to the gauge type which is compatible with the prometheus text format.
I bumped up the agent from version 1.25.0 to 2.0.0 and the issue is gone.
